superset/viz.py

```diff
@@ -341,6 +341,9 @@ def process_query_filters(self) -> None:
         utils.convert_legacy_filters_into_adhoc(self.form_data)
         merge_extra_filters(self.form_data)
         utils.split_adhoc_filters_into_base_filters(self.form_data)
+        # Normalize various filters, and then remove the original filters
+        self.form_data.pop("adhoc_filters", None)
+        self.form_data.pop("extra_filters", None)
```

Comment:
As we're mutating `self.form_data` in the above function calls, I wonder if this logic should be moved there?

Reply:
Is the logic of changing the filter here?

Reply:
I want to put all of modifying the filter together